src/ProxyWrappers.cxx

```diff
@@ -89,10 +89,42 @@ static PyObject* CreateNewCppProxyClass(Cppyy::TCppScope_t klass, PyObject* pyba
 }
 
 static inline
-void AddPropertyToClass(PyObject* pyclass,
-    Cppyy::TCppScope_t scope, Cppyy::TCppScope_t data)
+void AddPropertyToClass(PyObject* pyclass, Cppyy::TCppScope_t scope,
+    Cppyy::TCppScope_t data, intptr_t additional_offset=0)
 {
-    CPyCppyy::CPPDataMember* property = CPyCppyy::CPPDataMember_New(scope, data);
+    if (!Cppyy::IsPublicData(data))
+        return;
+
+    // enum datamembers (this in conjunction with previously collected enums above)
+    if (Cppyy::IsEnumType(Cppyy::GetDatamemberType(data)) &&
+        Cppyy::IsStaticDatamember(data)) {
+        // some implementation-specific data members have no address: ignore them
+        if (!Cppyy::GetDatamemberOffset(data))
+            return;
+
+        // two options: this is a static variable, or it is the enum value, the latter
+        // already exists, so check for it and move on if set
+        PyObject* eset = PyObject_GetAttrString(pyclass,
+            const_cast<char*>(Cppyy::GetFinalName(data).c_str()));
+        if (eset) {
+            Py_DECREF(eset);
+            return;
+        }
+
+        PyErr_Clear();
+    }
+
+    if (strstr(Cppyy::GetDatamemberTypeAsString(data).c_str(), "anonymous struct at") ||
+        strstr(Cppyy::GetDatamemberTypeAsString(data).c_str(), "anonymous union at")) {
+        std::vector<Cppyy::TCppScope_t> datamembers = Cppyy::GetDatamembers(Cppyy::GetTypeScope(data));
+        for (auto &datamember: datamembers) {
+            // properties (aka public (static) data members)
+            AddPropertyToClass(pyclass, scope, datamember,
+                                additional_offset + Cppyy::GetDatamemberOffset(data));
+        }
+    }
```

Comment:
Recursion is used here.

I can undo the changes in `BuildScopeProxyDict`, but then I will have to duplicate the same things in `AddPropertyToClass` for recursion to work.